File: LeetCode_Marathons/LCM4/Medium/Coin_Change_II/a.cpp
class Solution {
public:
    int change(int amount, vector<int>& coins) {
        int* dp = new int[amount + 1];
        for (int i = 0; i <= amount; i++) dp[i] = 0;
        dp[0] = 1;
        for (int coinAmount : coins) {
            for (int i = 0; i <= amount; i++) {
                if (i - coinAmount >= 0) {
                    dp[i] += dp[i - coinAmount];
                }
            }
        }
        int ans = dp[amount];
        delete[] dp;
        return ans;
    }
};

File: LeetCode_Marathons/LCM4/Medium/Coin_Change_II/a.go
func change(amount int, coins []int) int {
    dp := make([]int, amount + 1)
    dp[0] = 1
    for _, coinAmount := range coins {
        for i := 0; i <= amount; i++ {
            if i - coinAmount >= 0  {
                dp[i] += dp[i - coinAmount]
            }
        }
    }
    return dp[amount]
}